File: component-acgnews/src/main/java/com/rabtman/acgnews/di/module/ISHNewsDetailModule.java
package com.rabtman.acgnews.di.module;

import com.rabtman.acgnews.mvp.contract.ISHNewsDetailContract;
import com.rabtman.acgnews.mvp.model.ISHNewsDetailModel;
import com.rabtman.common.di.scope.ActivityScope;
import dagger.Module;
import dagger.Provides;

/**
 * @author Rabtman
 */
@Module
public class ISHNewsDetailModule {

  private ISHNewsDetailContract.View view;

  public ISHNewsDetailModule(ISHNewsDetailContract.View view) {
    this.view = view;
  }

  @ActivityScope
  @Provides
  ISHNewsDetailContract.View providerISHNewsDetailView() {
    return this.view;
  }

  @ActivityScope
  @Provides
  ISHNewsDetailContract.Model providerISHNewsDetailModel(ISHNewsDetailModel model) {
    return model;
  }
}

File: component-acgnews/src/main/java/com/rabtman/acgnews/di/module/ISHNewsDetailModule.kt
package com.rabtman.acgnews.di.module

import com.rabtman.acgnews.mvp.contract.ISHNewsDetailContract
import com.rabtman.acgnews.mvp.model.ISHNewsDetailModel
import com.rabtman.common.di.scope.ActivityScope
import dagger.Module
import dagger.Provides

/**
 * @author Rabtman
 */
@Module
class ISHNewsDetailModule(private val view: ISHNewsDetailContract.View) {
    @ActivityScope
    @Provides
    fun providerISHNewsDetailView(): ISHNewsDetailContract.View {
        return view
    }

    @ActivityScope
    @Provides
    fun providerISHNewsDetailModel(model: ISHNewsDetailModel): ISHNewsDetailContract.Model {
        return model
    }

}